src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -1,14 +1,33 @@
 package core.basesyntax.service;
 
+import core.basesyntax.Exception.NoValidDataException;
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        char []loginLength = user.getLogin().toCharArray();
+        char []passwordLength = user.getPassword().toCharArray();
+        for (User user1 : Storage.people) {
+            if (user.getLogin().equals(user1.getLogin())) {
+                throw new NoValidDataException("User with such login already exist");
+            }
+        }
+        if (loginLength.length < 6) {
+            throw new NoValidDataException("Login should have at least 6 characters");
+        }
+        if (passwordLength.length < 6) {
+            throw new NoValidDataException("Password should have at least 6 characters");
+        }
+        if (user.getAge() < 18) {
+            throw new NoValidDataException("User need to be at least 18 years old");
```

Comment:
Consider moving the hardcoded values like '6' for minimum login and password length, and '18' for minimum age to constant fields with descriptive names. This will make your code more readable and easier to maintain.

Reply:
Consider moving the hardcoded values like '6' for minimum login and password length, and '18' for minimum age to constant fields with descriptive names. This will make your code more readable and easier to maintain.